src/Service/PubliccodeService.php

```diff
@@ -0,0 +1,836 @@
+     * @param string $repositoryName  The fullname of the repository. /{owner}/{repository}
+     *
+     * @return string|null The updated logo with the download_url of the file contents with the path or null if not valid.
+     */
+    public function handleLogoFromGitlab(array $componentArray, Source $source, array $parsedLogo, string $repositoryName, string $repositoryId): ?string
+    {
+        // The url of the logo from the github repository always has /blob/{branch} or /uploads in the url.
+        // TODO: Check if this is also the case. If not this will be logged. Delete this comment if the log and the check if the log never appears.
+        $this->pluginLogger->warning('In this function we expect that a logo with host https://github.com always contains /blob/{branch} or /uploads in the URL. We need to think about whether we want to support this URL or whether we want to include it in the documentation (if it is not already the case)', ['open-catalogi/open-catalogi-bundle']);
+
+        // Check if /blob/ is not in the explodedPath. If so create a warning log.
+        if (str_contains($parsedLogo['path'], '/blob/') === false
+            && str_contains($parsedLogo['path'], '/uploads/') === true
+        ) {
+            // TODO: validate the /uploads url.
+            $this->pluginLogger->warning('TODO: The /uploads url isn\'t validated: '.$componentArray['logo'], ['open-catalogi/open-catalogi-bundle']);
+            return $componentArray['logo'];
+        }
+
+        // Check if /blob/ is in the explodedPath.
+        if (str_contains($parsedLogo['path'], '/uploads/') === false
+            && str_contains($parsedLogo['path'], '/blob/') === true
+        ) {
+            $this->pluginLogger->warning('In this function we expect that a logo with host https://gitlab.com always contains /blob/{branch} in the URL.', ['open-catalogi/open-catalogi-bundle']);
+
+            // Explode the logo path with the repositoryName so the organization an repository name will be removed from the path.
+            $explodedPath = explode($repositoryName, $parsedLogo['path'])[1];
+
+            // Explode the path with /.
+            // The first three items in the array is always:
+            // An empty string = 0, - = 1, blob = 2 and the branch = 3. This has to be removed from the path.
+            $explodedPath = explode('/', $explodedPath);
+
+            // Loop till the total amount of the explodedPath array.
+            $path = null;
+            // Set the branch default to main.
+            $branch = 'main';
+            for ($i = 0; $i < count($explodedPath); $i++) {
+                // If i is 0, 1, 2, 3 then nothing is done.
+                // The /-/blob/{branch} will not be added to the path.
+                if ($i === 0
+                    || $i === 1
+                    || $i === 2
+                ) {
```

Comment:
if (in_array($i, [0,1,2])) might look better and results in a 1 line if statement.
Or:

if ($i <= 2)
continue;

Reply:
Also, in your comment you say "// If i is 0, 1, 2, 3 then nothing is done." but you do something when $i = 3